servicelayer/process.py

```diff
@@ -47,14 +50,30 @@ def task_done(self):
         self.progress.mark_finished()
```

Comment:
I wonder if we should add an automatic test here to say: if done, remove the job/op?